code-challenges.test.js

```diff
@@ -43,11 +72,57 @@ var fullArr2 = ["hello", 7, 23, -823, false, 78, null, "67", 6, "number"]
 
 // b) Create the function that makes the test pass.
 
+//create a function and takes an array
+//create a for loop that loops through the array looking for typeOf number (try to use if you can use a higer order function)
+//once all the typeOf numbers are found, put it in order from least to leastToGreatest
+//return the new array of numbers put in order from least to greatest
+
+const leastToGreatest = (nums) => {
+  let oddNums = []
+  for(let i = 0; i < nums.length; i++){
+    if(typeof nums[i] === "number" && nums[i] % 2 !== 0){
+      oddNums.push(nums[i])
+    }
+  }
+  return oddNums.sort((a,b) => a - b)
+}
+console.log(leastToGreatest(fullArr1));
+console.log(leastToGreatest(fullArr2));
+
 
 
 // --------------------3) Create a function that takes in an array and returns an array of the accumulating sum. An empty array should return an empty array.
 
 // a) Create a test with expect statements for each of the variables provided.
+describe("sum", () => {
+  it("returns an array of accumlating sum", () => {
+    expect(arraySum(numbersToAdd1)).toEqual([2, 6, 51, 60])
+    expect(arraySum(numbersToAdd2)).toEqual([0, 7, -1, 11])
+    expect(arraySum(numbersToAdd3)).toEqual([])
+  })
+})
+
+// b) Create the function that makes the test pass.
+
+//create a function that takes in an array
+//the array is iterated through (or try to use a higher order function to shorten the steps)
+//the previous index is compared to the next index
+//those two index are added and that number is saved and printed (as an array)
+//the saved num is add to the next num in the arrays and printed as well
+//continue to do this until the array is over
+
+const arraySum = (arrayNum) => {
+  let newSum = []
+  for(let i = 1; i < arrayNum.length; i++){
+    if(i === 0){
```

Comment:
When I get rid of the conditional get get a few of the values being summed up. You're on the right track but I think this conditional is getting in the way. What did your tests say?